src/schema/comments.ts

```diff
@@ -284,6 +303,139 @@ export interface GQLUserCommentsArgs extends ConnectionArguments {
   userId: string;
 }
 
+interface MentionedUser {
+  id: string;
+  username?: string;
+}
+
+const getVerifiedMentions = async (
+  con: ORMConnection | EntityManager,
+  usernames: string[],
+  userId: string,
+): Promise<MentionedUser[]> =>
+  con
+    .getRepository(User)
+    .find({ where: { username: In(usernames), id: Not(userId) } });
+
+const getMentions = async (
+  con: ORMConnection | EntityManager,
+  content: string,
+  userId: string,
+): Promise<MentionedUser[]> => {
+  const words = content.split(' ');
+  const result = words.reduce((list, word) => {
+    if (word[0] !== '@' || word.length === 1) {
+      return list;
+    }
+
+    return list.concat(word.substring(1));
+  }, []);
+
+  if (result.length === 0) {
+    return [];
+  }
+
+  return getVerifiedMentions(con, result, userId);
+};
+
+const saveMentions = (
+  transaction: ORMConnection | EntityManager,
+  commentId: string,
+  commentByUserId: string,
+  users: MentionedUser[],
+) => {
+  const ids = users.map((user) => user.id);
+  transaction
+    .createQueryBuilder()
+    .delete()
+    .from(CommentMention)
+    .where({ commentId })
+    .andWhere({ commentByUserId })
+    .andWhere({ mentionedUserId: Not(In(ids)) })
+    .execute();
```

Comment:
Actually I agree with what you mentioned in the comments that it might better to not delete from the table